<launch>
	<include file="$(find sdu_pichi_demining)/launch/includes/decision_making.launch"/>
	
	<!-- from action -->
	<group ns="platform_executors">
		<node pkg="position_action_server" name="position_planner" type="positionGoalActionServer.py" output="screen">
			<param name="cmd_vel_topic" value="/fmSignals/cmd_vel" />
			<param name="odom_topic" value="fmKnowledge/odom" />
			<param name="max_linear_velocity" value="1"/>
			<param name="max_angular_velocity" value="4.0"/>
			<param name="straight_line" value="True"/>
			<param name="straight_line_angle_error" value="0.5"/>
		</node>
		
		<node pkg="action_primitives" name="turn_executor" type="make_turn.py" output="screen">
		     <param name="name" value="make_turn" />
		     <param name="odom_frame" value="/odom" />
		     <param name="base_frame" value="/base_footprint" />
		     <param name="cmd_vel_topic" value="/fmSignals/cmd_vel" />
		</node>
	</group>
	
	<!-- from perception -->
		 <node pkg="occupancy_grid_updater" type="grid_updater.py" name="map" output="screen">
 		  	<param name="sensor_width" value="2.0" />
	      	<param name="sensor_length" value="1.0" />
	      	<param name="sensor_offset_x" value="0.6" />
	      	<param name="sensor_offset_y" value="0.0" />
	      	<param name="sensor_outerrange" value="0.1" />
	      	<param name="resolution" value="0.1" /> <!-- Size of cell in [m] -->
	      	<param name="perios" value="0.1" /> <!-- Frequency of node in [Hz] -->
		</node>
		
		<node pkg="occupancy_grid_updater" type="grids_combined_node" name="combiner" output="screen"/>

 	<node pkg="stage" type="stageros" name="simulator" args="$(find sdu_pichi_demining)/world/pichi_demining.world"/>
	<node pkg="twist_stamped_to_twist" name="converter" type="cmd_vel_converter.py"/>
	
	<node pkg="wads_implement" name="WADSImplementSimulator" type="wads_implement_node">
	      <param name="sensor_width" value="2.0" />
	      <param name="sensor_length" value="1.0" />
	      <param name="sensor_offset_x" value="0.6" />
	      <param name="sensor_offset_y" value="0.0" />
	      <param name="sensor_gain" value="25" />
	      <param name="sensor_outerrange" value="0.0" />
	      <param name="sensor_max" value="5.0" />
	      <param name="sensor_resolution" value="1024" />
	      <param name="subscriberTopic" value="/base_scan" />
	      <param name="publisherTopic" value="/wads" />
	</node>
	
	<node pkg="wads_implement" name="WADSVisualizer" type="wads_implement_visualizer">
	      <param name="sensor_width" value="2.0" />
	      <param name="sensor_length" value="1.0" />
	      <param name="sensor_offset_x" value="0.6" />
	      <param name="sensor_offset_y" value="0.0" />
	      <param name="sensor_outerrange" value="0.1" />
	      <param name="sensor_max" value="5.0" />
	      <param name="subscriberTopic" value="/wads" />
	      <param name="publisherTopic" value="/marker" />
	      <param name="frame_id" value="/base_link" />
	</node>

</launch>
